lib/register_client.rb

```diff
@@ -17,37 +20,44 @@ def initialize(register, phase, config_options)
     def get_entries
       get_data[:entries][:user]
     end
+    filter_and_paginate :get_entries
 
     def get_records
       get_data[:records][:user].map { |_k, v| v.last }
     end
+    filter_and_paginate :get_records
 
-    def get_metadata_records
-      get_data[:records][:system].map { |_k, v| v.last }
+    def get_records_with_history
+      get_data[:records][:user]
     end
+    filter_and_paginate :get_records_with_history
 
-    def get_field_definitions
-      get_metadata_records.select { |record| record[:key].start_with?('field:') }
+    def get_current_records
+      get_records_no_pagination.select { |record| record[:item]['end-date'].nil? }
```

Comment:
If we are certain `:item` must be there, using `record.fetch(:item)['end-date']` gives you a better error if it's not there. You get a `KeyError: key not found: :x` rather than `NoMethodError: undefined method `[]' for nil:NilClass`.

Reply:
The strategy applies to any hash key you consume assuming it will always be there.